lua/luasnip/util/environ.lua

```diff
@@ -39,14 +39,14 @@ function Environ.is_table(var_fullname)
 	return nmsp.is_table(varname)
 end
 
-function Environ:new(pos, o)
+function Environ:new(meta, o)
 	o = o or {}
 	setmetatable(o, self)
 
 	for ns_name, ns in pairs(namespaces) do
 		local eager_vars = {}
 		if ns.init then
-			eager_vars = ns.init(pos)
+			eager_vars = ns.init(meta)
```

Comment:
Mhmm, what do you think of changing the table passed to `init` to
```lua
{
  pos = {row,col},
  trigger = ...
  captures = {...}
}
```
?
We could set `meta[1], meta[2]` (btw what's the mnemonic here? metadata?) to `row`, `col` to stay backward compatible

Reply:
> btw what's the mnemonic here? metadata? 
It's either that or a reference to the effects of  Meta over the environment :interrobang: 




> We could set `meta[1], meta[2]`

Do you think then I should do that and keep `meta[1] & meta[2]` just as an undocumented non-breaking change protection?

Reply:
> > btw what's the mnemonic here? metadata?
> 
> It's either that or a reference to the effects of Meta over the environment :interrobang:

:laughing:
Mhmm, what `data` is this the `metadata` to though?
(Just to be clear, I'm not worried about the name of the variable itself, more about what we call it in the documentation. I think `meta` is a bit confusing (+negatively connotated :P), just `info` maybe?)

> > We could set `meta[1], meta[2]`
> 
> Do you think then I should do that and keep `meta[1] & meta[2]` just as an undocumented non-breaking change protection?

Yes, undocumented :+1::+1: